Add twist_mux node

matthewia94 opened the pull request:
The twist_mux node is a node which takes in multiple twist messages and
publishes only a single twist message to the motor controller on the
"cmd_vel" topic.  The node currently uses the buttons defined in the
constants.h file.

Closes #3

matthewia94 committed:
Add twist_mux node

The twist_mux node is a node which takes in multiple twist messages and
publishes only a single twist message to the motor controller on the
"cmd_vel" topic.  The node currently uses the buttons defined in the
constants.h file.

islamelnabarawy commented on ros_ws/src/common/constants.h:
```diff
@@ -22,15 +38,27 @@ const int X_AXIS_R_STICK = 2;
+/**
+ * @brief The topic containing twist messages to be sent to the motor controller
+ */
+const std::string CMD_VEL_TOPIC = "cmd_vel";
```
Let's rename CMD_VEL_TOPIC to be something more generic so we wouldn't be locked into using the particular topic called "cmd_vel". Maybe CONTROL_TOPIC or something similar...

islamelnabarawy commented on ros_ws/src/control/src/control/twist_mux.cpp:
```diff
@@ -0,0 +1,57 @@
+    cmd_vel.linear.z = msg->linear.z;
+    cmd_vel.angular.x = msg->angular.x;
+    cmd_vel.angular.y = msg->angular.y;
+    cmd_vel.angular.z = msg->angular.z;
```
You don't have to manually copy every single member. Just doing the following assignment should be enough:

cmd_vel = *msg;

islamelnabarawy commented on ros_ws/src/control/src/control/twist_mux.cpp:
```diff
@@ -0,0 +1,57 @@
+    cmd_vel.linear.z = 0;
+    cmd_vel.angular.x = 0;
+    cmd_vel.angular.y = 0;
+    cmd_vel.angular.z = 0;
```
Similarly, I believe assigning an empty Twist will accomplish the same thing here:

cmd_vel = geometry_msgs::Twist();

islamelnabarawy commented on ros_ws/src/control/include/control/twist_mux.h:
```diff
@@ -0,0 +1,104 @@
+        standby,        //!< The robot is not moving, the same as a software e-stop
+        teleop,         //!< The robot is currently being controlled by remote control
+        autonomous      //!< The robot is driving itself
+    };
```
This (unscoped) enum will pollute the global namespace with the names of its members.

Please see [the ROS C++ Style Guide section on Enumerations](http://wiki.ros.org/CppStyleGuide#Enumerations) for the proper way to define enums without polluting the global namespace.

matthewia94 committed:
Rename CMD_VEL_TOPIC

The constant used to transmit twist messages to the motor controllers
has been renamed from CMD_VEL_TOPIC to CONTROL_TOPIC to make it more
general.

matthewia94 committed:
Add Control Namespace

The control namespace should avoid issues with namespace pollution
specifically with the mode enum.

matthewia94 committed:
Change twist subscriber function

Instead of using two subscribers for two different topics there is now
only one subscriber which changes the topic it subscribes to based on
joy inputs.

matthewia94 commented:
This also addresses issue #4
